python/paddle/fluid/tests/unittests/test_elementwise_sub_op.py

```diff
@@ -16,239 +16,396 @@
+        if self.dtype == np.float16:
+            return
+        self.check_grad(['Y'], 'Out', no_grad_set=set("X"))
 
     def test_check_grad_ingore_y(self):
-        self.check_grad(
-            ['X'], 'Out', max_relative_error=0.005, no_grad_set=set('Y'))
+        # TODO(wangzhongpu): support mkldnn op in dygraph mode
+        if self.dtype == np.float16:
+            return
+        self.check_grad(['X'], 'Out', no_grad_set=set('Y'))
+
+    def init_input_output(self):
+        self.x = np.random.uniform(0.1, 1, [13, 17]).astype(self.dtype)
+        self.y = np.random.uniform(0.1, 1, [13, 17]).astype(self.dtype)
+        self.out = np.subtract(self.x, self.y)
+
+    def init_dtype(self):
+        self.dtype = np.float64
+
+    def init_axis(self):
+        self.axis = -1
+
+
+@OpTestTool.skip_if(not core.is_compiled_with_cuda(),
+                    "core is not compiled with CUDA")
+class TestFP16ElementwiseSubOp(TestElementwiseSubOp):
+    def init_dtype(self):
+        self.dtype = np.float16
+
+    def test_check_output(self):
+        # TODO(wangzhongpu): support mkldnn op in dygraph mode
+        if core.is_compiled_with_cuda():
+            place = core.CUDAPlace(0)
+            if core.is_float16_supported(place):
+                self.check_output_with_place(place)
+
+
+@OpTestTool.skip_if(True, "Grad not yet implemented")
+class TestElementwiseSubOp_scalar(TestElementwiseSubOp):
+    def init_input_output(self):
+        self.x = np.random.rand(2, 3, 4).astype(self.dtype)
+        self.y = np.random.rand(1).astype(self.dtype)
+        self.out = self.x - self.y
 
 
 @skip_check_grad_ci(
     reason="[skip shape check] Use y_shape(1) to test broadcast.")
-class TestElementwiseSubOp_scalar(TestElementwiseOp):
-    def setUp(self):
-        self.op_type = "elementwise_sub"
-        self.inputs = {
-            'X': np.random.rand(10, 3, 4).astype(np.float64),
-            'Y': np.random.rand(1).astype(np.float64)
-        }
-        self.outputs = {'Out': self.inputs['X'] - self.inputs['Y']}
+class TestFP16ElementwiseSubOp_scalar(TestFP16ElementwiseSubOp):
+    def init_input_output(self):
+        self.x = np.random.rand(2, 3, 4).astype(self.dtype)
+        self.y = np.random.rand(1).astype(self.dtype)
+        self.out = self.x - self.y
 
 
-class TestElementwiseSubOp_Vector(TestElementwiseOp):
-    def setUp(self):
-        self.op_type = "elementwise_sub"
-        self.inputs = {
-            'X': np.random.random((100, )).astype("float64"),
-            'Y': np.random.random((100, )).astype("float64")
-        }
-        self.outputs = {'Out': self.inputs['X'] - self.inputs['Y']}
+@OpTestTool.skip_if(True, "Grad not yet implemented")
+class TestElementwiseSubOp_scalar2(TestElementwiseSubOp):
+    def init_input_output(self):
+        self.x = np.random.rand(2, 3, 4).astype(self.dtype)
+        self.y = np.random.rand(1, 1).astype(self.dtype)
+        self.out = self.x - self.y
 
 
```

Comment:
```suggestion
@OpTestTool.skip_if(True, "Grad not yet implemented")
```

You have disabled the upper test, now CI fails on that one having no FP16 kernel

Reply:
And from what I can see in CI, double grad is causing problems, not just single grad
![image](https://user-images.githubusercontent.com/62569058/133980470-fc719d5c-6eb4-4f84-a90a-2cc21090f00d.png)

Reply:
Most of the elementwise_sub tests will be removed anyway, because they were copied from elementwise_add and it turned out that most are just not supported. Right now I'm working on making sure that grad tests work with the mkldnn implementation, because there were disabled until now.